pkg/kubelet/container/runtime.go

```diff
@@ -185,14 +185,38 @@ type Image struct {
 	Size int64
 }
 
+type Mount struct {
+	// Name of the volume mount.
+	Name string
+	// Path of the mount within the container.
+	ContainerPath string
+	// Path of the mount on the host.
+	HostPath string
+	// Whether the mount is read-only.
+	ReadOnly bool
+}
+
+type PortMapping struct {
+	// Name of the port mapping
+	Name string
+	// Protocol of the port mapping.
+	Protocol api.Protocol
+	// The port number within the container.
+	ContainerPort int
+	// The port number on the host.
+	HostPort int
+	// The host IP.
+	HostIP string
+}
+
 // RunContainerOptions specify the options which are necessary for running containers
 type RunContainerOptions struct {
-	// The environment variables, they are in the form of 'key=value'.
-	Envs []string
+	// The environment variables map.
+	Envs map[string]string
```

Comment:
Hmm some people have env vars that depend on other env vars so the order is probably important.